### PR TITLE
dApp: `AcreTVLProgress` component

### PR DESCRIPTION
**Pull request by kpyszkowski**
Closes: #822

### Changes:
- Removed the use of term `season` - we don't have seasons any more
- Implemented `AcreTVLProgress` component
- Updated the TVL hook
- Updated styles of `Progress` component

### Preview:

image

### Note:
To start with latest layout changes I set the origin branch to [`ledger-live-updates`](https://github.com/thesis/acre/tree/ledger-live-updates)

### CODE REVIEW

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (kkosiorowska) [inline dapp/src/pages/DashboardPage/AcreTVLProgress.tsx:39]: I think these values should depend on `TOTAL_VALUE_LOCKED_CAP`. A quick solution that comes to my mind:

```
export const useTVL = () => {

  ...
  const totalValue = bigIntToUserAmount(
    TOTAL_VALUE_LOCKED_CAP,
    BITCOIN.decimals,
  )

  return { progress, value, isCapExceeded, totalValue }
}

```

```
const UNIT = 50

const { totalValue } = totalValueLocked
const scale = Array.from({ length: totalValue / UNIT },(_, idx) => totalValue - idx * UNIT).reverse()
```

Let me know what you think and if you may have any other solution.

**Review by kkosiorowska — COMMENTED** [review]
Looks awesome 😍 I left minor comments.

File: pnpm-lock.yaml
Comment (kpyszkowski) [reply pnpm-lock.yaml:1]: Those changes are only in file formatting. I use the latest version of pnpm - maybe they did some update to lockfile parser.

File: dapp/src/assets/icons/BoltFilled.tsx
Comment (kpyszkowski) [reply dapp/src/assets/icons/BoltFilled.tsx]: Ref commit: 1953d7d0

File: dapp/src/assets/icons/index.ts
Comment (kpyszkowski) [reply dapp/src/assets/icons/index.ts]: Ref commit: 1953d7d0

File: dapp/src/hooks/index.ts
Comment (kpyszkowski) [reply dapp/src/hooks/index.ts]: Ref commit: 1953d7d0

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx:39]: I discussed it with @nkuba.
The conclusion was to hardcode values.
Context [here](https://discord.com/channels/1079490158639976609/1297889075050582069/1300486504128446628)

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (nkuba) [inline dapp/src/pages/DashboardPage/AcreTVLProgress.tsx:30]: Currently, we show `244.6570`. Can we reduce the number of decimals? How about we show two decimals?

File: dapp/src/utils/acreApi.ts
Comment (kpyszkowski) [reply dapp/src/utils/acreApi.ts:109]: Ref commit:  a65743cb

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx:41]: Ref commit: 880130a9

**Comment by kpyszkowski** [comment]
@nkuba 
As requested, updated the cap to 1250.
Ref commit: e586c639

File: dapp/src/constants/staking.ts
Comment (r-czajkowski) [inline dapp/src/constants/staking.ts:4]: Non-blocking comment: what do you think about passing this value as an env variable? We could change the cap w/o updating the code.

File: dapp/src/components/shared/ProgressBar.tsx
Comment (r-czajkowski) [inline dapp/src/components/shared/ProgressBar.tsx:12]: Why greater than `2`? This is a shared component so are we sure it will be ok for other use cases?

File: dapp/src/hooks/useTVL.ts
Comment (r-czajkowski) [inline dapp/src/hooks/useTVL.ts:24]:
```suggestion
  const progress =
    isCapExceeded ? 100 :  Math.floor((totalAssets / TOTAL_VALUE_LOCKED_CAP) * 100)
```

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (r-czajkowski) [inline dapp/src/pages/DashboardPage/AcreTVLProgress.tsx:21]: Can't it just be a const value?

```suggestion
const getSteps =
```

File: dapp/src/theme/Progress.ts
Comment (r-czajkowski) [inline dapp/src/theme/Progress.ts:14]: Let's define this color. If it's only used in the `Progress` component we can declare a const value but outside of the function - for example at the beginning of this file.

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (nkuba) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx]: Let's assume we set the cap to 1250, so we have a step every 250. When we hit it and update the `TOTAL_VALUE_LOCKED_CAP` to 1600, the new progress bar will have a step every 320. So, the previous steps and the goal of 1250 are no longer displayed. This may be confusing as we don't maintain the historic goals.

I preferred hardcoding the steps, as depending on the change in the cap, we need to consider the steps to keep them consistent across updates.

We can keep the current automated calculation, but we need to be aware that we need to revisit the steps in case of an update in the future. It is worth adding a comment in the code.

Comment (nkuba) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx]: Let's assume we set the cap to 1250, so we have a step every 250. When we hit it and update the `TOTAL_VALUE_LOCKED_CAP` to 1600, the new progress bar will have a step every 320. So, the previous steps and the goal of 1250 are no longer displayed. This may be confusing as we don't maintain the historic goals.

I preferred hardcoding the steps, as depending on the change in the cap, we need to consider the steps to keep them consistent across updates.

We can keep the current automated calculation, but we need to be aware that we need to revisit the steps in case of an update in the future. It is worth adding a comment in the code.

File: dapp/src/constants/staking.ts
Comment (nkuba) [reply dapp/src/constants/staking.ts:4]: We need to consider the steps when we update the cap. Please see https://github.com/thesis/acre/pull/823#discussion_r1827782305

If we change just the cap, without updating the steps, we won't maintain the historic goals.

File: dapp/src/hooks/useTVL.ts
Comment (nkuba) [inline dapp/src/hooks/useTVL.ts:13]: Total assets is a name specific to the [stBTC vault](https://github.com/thesis/acre/blob/main/solidity/contracts/stBTC.sol). But now we include in the TVL not only the stBTC vault total assets, but also deposits made to the [Multi-Asset Vault](https://github.com/thesis/acre/blob/main/solidity/contracts/AcreMultiAssetVault.sol).

Let's replace `totalAssets` with `tvlBtc` across the file. Let's also update the query keys factory.


```suggestion
    queryKey: [...acreKeys.tvlBtc()],
```

File: dapp/src/constants/staking.ts
Comment (kpyszkowski) [reply dapp/src/constants/staking.ts:4]: Discussed with Kuba. Decided to hardcode it as a constant. It won't be changed that often.

File: dapp/src/hooks/useTVL.ts
Comment (kpyszkowski) [reply dapp/src/hooks/useTVL.ts]: Refactored for consistency: f4eb4bd1
You called the endpoint "statistics" so I followed the convention.

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx]: Ref commit: e985c20c

File: dapp/src/hooks/useTVL.ts
Comment (kpyszkowski) [reply dapp/src/hooks/useTVL.ts]: Ref commit: 6d9e39ba

File: dapp/src/components/shared/ProgressBar.tsx
Comment (kpyszkowski) [reply dapp/src/components/shared/ProgressBar.tsx]: Ref commit: c51c695c

**Review by r-czajkowski — APPROVED** [review]
LGTM! I think we can hold off on merging until we merge the https://github.com/thesis/acre/pull/708 into `main`.

File: dapp/src/pages/DashboardPage/AcreTVLProgress.tsx
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/AcreTVLProgress.tsx]: The number of steps on the progress bar is currently parametrized with a variable.

```ts
const STEP_COUNT = 5
```
- 4 steps: image

- 5 steps: image

- 6 steps: image

In my opinion it's enough control for now.  We can adjust it later in case more control is needed (like irregular steps you mentioned).

**Review by nkuba — APPROVED** [review]

**Pull request merged**
